pkg/lifecycle-poc/funnel/processor.go

```diff
@@ -88,38 +88,42 @@ func (t *ProcessorTask) Do(ctx context.Context, b *Batch) error {
 	}
 	t.metrics.Observe(len(recsOut), start)
 
+	// Map active records back to their original indices in the batch, but only
+	// if we have filtered records. Otherwise, we can just use the original indices.
+	var activeIndices []int
+	if b.filterCount > 0 {
+		activeIndices = make([]int, 0, len(b.records))
```

Comment:
Shouldn't this be the length of `b.filterCount` instead?

```suggestion
		activeIndices = make([]int, 0, len(b.filterCount))
```

Reply:
Unrelated, as someone who is not very familiar with the processor code, I feel this `Do` method could benefit from a comment and/or a refactor, as I see it is kind of lengthy.